### PR TITLE
Ftr: compatible with lower versions registering "default." keys

### LINKED ISSUE #1442: 调2.6.x版本 生产者default.version字段不兼容，导致匹配不到
// ServiceKey gets a unique key of a service.
func (c *URL) ServiceKey() string {
	return ServiceKey(c.GetParam(constant.INTERFACE_KEY, strings.TrimPrefix(c.Path, "/")),
		c.GetParam(constant.GROUP_KEY, ""), c.GetParam(constant.VERSION_KEY, ""))
}

c.GetParam(constant.VERSION_KEY, "")

### PR DESCRIPTION
**Pull request by ChangedenCZD**
**What this PR does**:
Reference Solutions: [https://github.com/apache/dubbo/pull/5950](https://github.com/apache/dubbo/pull/5950)

**Which issue(s) this PR fixes**:

Fixes #1442

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: common/url.go [hunk 1]
```diff
@@ -259,6 +259,16 @@ func NewURL(urlString string, opts ...Option) (*URL, error) {
 		return &s, perrors.Errorf("url.ParseQuery(raw url string{%s}),  error{%v}", serviceUrl.RawQuery, err)
 	}
 
+	// compatible "default." prefix
+	for key, value := range s.params {
+		if strings.Index(key, constant.DEFAULT_KEY_PREFIX) == 0 && len(value) > 0 {
```
Comment (zouyx) [inline common/url.go:264]: Should you use `strings.HasPrefix`?

File: protocol/invocation/rpcinvocation.go [hunk 2]
```diff
@@ -165,6 +165,13 @@ func (r *RPCInvocation) SetAttachments(key string, value interface{}) {
 	if r.attachments == nil {
 		r.attachments = make(map[string]interface{})
 	}
+	// compatible "default." prefix
+	if strings.Index(key, constant.DEFAULT_KEY_PREFIX) == 0 {
```
Comment (zouyx) [inline protocol/invocation/rpcinvocation.go:169]: Should you use `strings.HasPrefix`?

File: common/url_test.go [hunk 3]
```diff
@@ -421,3 +421,11 @@ func TestCompareURLEqualFunc(t *testing.T) {
 func CustomCompareURLEqual(l *URL, r *URL, execludeParam ...string) bool {
 	return l.PrimitiveURL == r.PrimitiveURL
 }
+
+func TestURLWithLowerVersion(t *testing.T) {
```
Comment (zouyx) [inline common/url_test.go:425]: I think your test case has not cover all scene.
You should provide a new test case about `default.version` replace `version`.

File: common/url.go (hunk 1 above)
Comment (ChangedenCZD) [reply common/url.go]: Done

File: protocol/invocation/rpcinvocation.go (hunk 2 above)
Comment (ChangedenCZD) [reply protocol/invocation/rpcinvocation.go]: Done

**Review by cjphaha — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by zhaoyunxing92 — APPROVED** [review]

**Pull request merged**